### ISSUE TITLE
Request

### ISSUE THREAD

**Issue by mikhoul** [Quicksaver/Puzzle-Toolbars#94]
Could it be possible to add to the "**_Puzzle toolbars_**"  bookmarks and Bookmark Folders ?  

I'd like to make a toolbar for my Dev/JS project with the devtools I use the most and ALSO including the shortcuts to the web tools I use the most and  have it all packed in a nifty place.  :smile_cat: 

Best Regards ! :octocat:

**Comment by Quicksaver** [comment]
You could use the top toolbar and move your bookmarks stuff in there. Does this not work for you?

**Comment by mikhoul** [comment]
It's already full ---> http://i.imgur.com/9iOZW89.png

Also I don't see the Top bar from "Puzzle toolbars" , I even send some tools but can't find them ?

Regards :smiley_cat:

**Comment by Quicksaver** [comment]
You have to go to the add-on's preferences and enable it first, it's disabled by default. Unless you already did?

**Comment by Quicksaver** [comment]
Nevermind, I misread. So is that top toolbar the one from Puzzle Bars or the Bookmarks toolbar that comes with Firefox? You can have both even.

**Comment by mikhoul** [comment]
Both are from Firefox, to be sure we talk about the same thing the toolbars highlighted in yellow are from Firefox : http://i.imgur.com/Qnhn2FN.png

I'm not able to make appear the top bar from "Puzzle Bars" ???  Other bars: bottom, corner, lateral, I have no problem to find it but for the top bar I don't catch how to make it appear ?


Let me know if you need more info ! :smiley_cat:

**Comment by Quicksaver** [comment]
Alright, let's start with this, is it not listed in the menu when you right-click any of the toolbars? What about if you center Customize, is it in the menu that opens when you click the "Show / Hide Toolbars" button at the bottom? And if you go to the add-on's preferences and set a keyboard shortcut for it (any at all, just as a test), can you toggle it using that shortcut at least?

If none of those work for you, something could be preventing it from either initializing or just appearing. Could you try disabling all your themes and add-ons, except for Puzzle Toolbars of course, as a test and see if you can use it then? If it is a conflict with another add-on, I'll need to know which one in order to fix it.

**Comment by mikhoul** [comment]
I found the culprit ! :smiley_cat: 

The only addon that was "playing" with the chrome of Firefox was [CTR](https://addons.mozilla.org/firefox/addon/classicthemerestorer/) so as soon as I disabled it it was working fine.

Sadly with CTR and my Tabs below it doesn't play fine, the Puzzle Bars button is hidden behind the "+" button to add a new tab: http://i.imgur.com/MQBUcUw.png and http://i.imgur.com/FGTR7Ig.png

Here's my settings file if you want to try it with CTR:  [CTRpreferences.zip](https://github.com/Quicksaver/Puzzle-Toolbars/files/524739/CTRpreferences.zip)

I've made a test with [Custom Buttons](https://addons.mozilla.org/firefox/addon/classicthemerestorer/) and I was able to drag and drop them to the "puzzle toolbar" since they have their own context menu. 😊

-=-=-=-=-=-=-=-=- 

I'd really like to be able to put bookmarks beside tools to make a toolbar with only tools for Programming and since a lot of resources/tools I use are online bookmarks beside various Devtools would be useful.  http://i.imgur.com/HbfgHC8.png  :octocat: 

Let me know if you think it would be possible. 

In meantime Have a nice day !  :partly_sunny:

**Comment by Quicksaver** [comment]
I'll look into fixing compatibility with CTR as soon as I can, it should be fairly straightforward. By the way, CTR has its own extra toolbars at the top I believe (somewhere in its preferences you can define extra toolbars), you should be able to use those as well.

> I'd really like to be able to put bookmarks beside tools to make a toolbar with only tools for Programming 

Actually, those bookmarks are inside a "Bookmarks item" which can be moved at will in customize mode; that item itself is currently in the "Bookmarks toolbar" (they're different things!). So basically you can drag the bookmarks into any other toolbar (I never tried putting them in the lateral bar, so I'm not sure how they will look there), and you can drag other buttons besides your bookmarks as well. Have you tried doing this in customize mode? Is that what you're looking for or did I misunderstood?

**Comment by mikhoul** [comment]
With CTR I was not able to reach the puzzle button since there was another CSS layer over but in meantime I've put a shortcut.

For the bookmarks I'm not sure we talk about the same thing  😏

Here's a quick screencast when I try to drag and drop bookmarks on the puzzle toolbars, I tried the same in customize mod but sadly with the same results:  http://screencast-o-matic.com/watch/cD6l2Cjacu

**Comment by Quicksaver** [comment]
You can only drag them in customize view. Open the main menu (the three bars button, you have it at the top, after the smiley button after the stylish button), then click on customize. Now you can drag the bookmarks item anywhere else, and you can also drag other buttons next to it.

Check out the video at https://support.mozilla.org/kb/customize-firefox-controls-buttons-and-toolbars

**Comment by mikhoul** [comment]
Yes I know for the "Bookmarks TOOL button" but I was asking for the individual bookmarks (like Googgle.ca or github.com etc) to be pinned/dragged on the "puzzle toolbar" mixed with others buttons. 

Like the "[Bookmarks Toolbar](https://support.cdn.mozilla.net/media/uploads/gallery/images/2014-03-26-09-00-53-b8cf84.png)", being able to mix Tool buttons and bookmarks on the same toolbar. :octocat:

**Comment by Quicksaver** [comment]
Oh, to mix them all together. I'm sorry, I hadn't understood that part. Unfortunately no, that's just not possible without some serious spelunking into its code. It's an interesting concept, but definitely not one I would dwell into any time soon, especially with the whole WebExtensions thing going on at the moment.

**Comment by mikhoul** [comment]
I understand it take lot of time to maintain all those useful programs :smile_cat: 

Thanks again for the help ! :octocat:

**Comment by Quicksaver** [comment]
You can leave it open. If I have time some day, I may look into it. ;)

**Comment by Quicksaver** [comment]
And there's the CTR I need to check.

**Comment by mikhoul** [comment]
👍

**Comment by Quicksaver** [comment]
> the Puzzle Bars button is hidden behind the "+" button to add a new tab

About this part, the puzzle icon appears above the plus icon for me, and not behind it. Does that also happen for you if you disable all your other add-ons except for Puzzle Bars and CTR?

**Comment by Quicksaver** [comment]
And also please try [beta version 2.1.15b1](https://github.com/Quicksaver/Puzzle-Toolbars/releases/tag/v2.1.15b1) and let me know if it makes any difference.

**Comment by mikhoul** [comment]
Hi,

I used a cleaner profile to test it and I've been able to replicate the issue.

When the icon of TAB GROUP is located on the TAB BAR at right  it work fine BUT as soon the icon is removed from the TAB BAR the Top Puzzle Bar Icon don't show up.

It's very strange since the 2 addons are not really related. I discovered that because on the new profile I had to install both addons and the default position of the the TAB BAR Icon was on the TAB BAR, so I tried to move it to the toolbar like my main profile and from now I was not able to bring back the Top puzzle bar.

Here's a quick screencast to help you to have a better understanding of what I try to explain :smile_cat: 

http://screencast-o-matic.com/watch/cD6rIcjC1w 


Let me know if you need more information !

Regards :octocat:

**Comment by Quicksaver** [comment]
Was that with the beta I linked above?

**Comment by mikhoul** [comment]
Yes it was with the beta :smile_cat:

**Comment by Quicksaver** [comment]
So the problem is that the empty area of the tabs is non-reactive at all. This is to allow dragging the window for instance, it's not something I'm willing to change, as I have no idea what else I would break by doing it (I'm not sure I could change it easily anyway). Basically you can't get the puzzle piece to appear because it would have to react to the area where it should be, which it can't. The groups button (or any other) is a reactive node itself, which is why it makes it work, the puzzle piece will be shown over reactive area now.

In [beta version 2.1.15b2](https://github.com/Quicksaver/Puzzle-Toolbars/releases/tag/v2.1.15b2) I made it so the puzzle piece is shown above the tabs, so you can at least move it to the left and it should always work there (because tabs are also reactive, yes I know, this whole thing sounds silly, Firefox likes to throw a curve ball here and there).

**Comment by mikhoul** [comment]
Thanks for the explanation ! :smile_cat:

**Comment by mikhoul** [comment]
Just to let you know with the beta version Right Now I'm able to  make it appear in the RIGHT side hovering on the "**+** Add new tab" button on the **_Right Upper side_** of the button. http://i.imgur.com/mUQEAnV.png

So it's enough for me to be able to use it on the right side ! :smile: 

Have a great Week-end ! :sun_with_face:

**Comment by Quicksaver** [comment]
Again, keeping it open for the bookmarks thing, this is why I prefer to keep different problems in different issues.
